We will have another simple example on two sample t test (pooled- when the variances are equal). But this time this is a **one sided t-test**<br>

1. In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file files_for_lab/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other?

In [38]:
import pandas as pd
import numpy as np
import scipy.stats as stats

new_machine=[42.1,41,41.3,41.8,42.4,42.8,43.2,42.3,41.8,42.7]
old_machine=[42.7,43.6,43.8,43.3,42.5,43.5,43.1,41.7,44,44.1]

data=pd.DataFrame(columns=['new_machine','old_machine'])
data['new_machine']=new_machine
data['old_machine']=old_machine

data

new_machine  old_machine
0         42.1         42.7
1         41.0         43.6
2         41.3         43.8
3         41.8         43.3
4         42.4         42.5
5         42.8         43.5
6         43.2         43.1
7         42.3         41.7
8         41.8         44.0
9         42.7         44.1

In [16]:
data_descr = data.describe().head(3)
data_descr

new_machine  old_machine
count    10.000000    10.000000
mean     42.140000    43.230000
std       0.683455     0.749889

In [24]:
data_descr.iloc[1,0]

42.14

##### Data:
H0: µ_new >= µ_old<br>
Ha" µ_new < µ_old<br>
a = 0.05

In [32]:
new_n = data_descr.iloc[0,0]
new_mean = data_descr.iloc[1,0]
new_std = data_descr.iloc[2,0]

old_n = data_descr.iloc[0,1]
old_mean = data_descr.iloc[1,1]
old_std = data_descr.iloc[2,1]

0.7174414416676962

In [33]:
# Calculate pool standard deviation:
p_std = np.sqrt(((new_n - 1)*new_std**2 + (old_n - 1)*old_std**2) / (new_n + old_n - 2))
p_std

0.7174414416676962

In [37]:
# Calculate t-value
t = (new_mean - old_mean)/(p_std * np.sqrt(2/10)) 
round(t, 3)

-3.397

In [39]:
# Find critical value, with 9 DoF, for one-tailed test
tt = stats.t.ppf(0.95, 9)
round(tt, 3)

1.833

* Calculated **t = -3.397**
* Critical **tt = 1.833**
<br>

We can reject the null ipothesis, or we can say that **there is enough evidence to reject the Null hypothesis, meaning that the new machine packs faster than the older.